lib/s3/models/BucketConfig.rb

```diff
@@ -98,6 +102,76 @@ def initialize(name, json = nil)
         end
       end
 
+      # Public: Populate this BucketConfig from the values in an AWS bucket.
+      #
+      # aws      - the aws resource
+      # cors     - a hash of the names of cors policies to the string value of those policies
+      # policies - a hash of the names of policies to the string value of those policies
+      #
+      # Returns the key names of the new policy or cors policy so they can be written
+      # to file immediately
+      def populate!(aws, cors, policies)
+        @region = aws.location
+        @grants = aws.acl.to_cumulus
+        @website = aws.website.to_cumulus
+        @logging = aws.logging.to_cumulus
+        @notifications = aws.notification.to_cumulus
+        @lifecycle = aws.lifecycle.to_cumulus
+        @versioning = aws.versioning.enabled
+        @replication = aws.replication.to_cumulus rescue nil
+        @tags = Hash[aws.tagging.safe_tags.map { |t| [t.key, t.value] }]
+
+        policy = aws.policy.policy_string
+        if policy and policy != ""
+          policy = JSON.pretty_generate(JSON.parse(policy))
+          if policies.has_value? policy
+            @policy_name = policies.key(policy)
+          else
+            @policy_name = "#{@name}-policy"
+            policies[@policy_name] = policy
+            @new_policy_key = @policy_name
+          end
+        end
+
+        cors_string = JSON.pretty_generate(aws.cors.rules.map(&:to_h))
+        if cors_string and !aws.cors.rules.empty?
+          if cors.has_value? cors_string
+            @cors_name = cors.key(cors_string)
+          else
+            @cors_name = "#{@name}-cors"
+            cors[@cors_name] = cors_string
+            @new_cors_key = @cors_name
+          end
+        end
+
+        [@new_policy_key, @new_cors_key]
```

Comment:
You can do return `@new_policy_key, @new_cors_key` and then assign with `policy, cors = populate!...`